staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -106,6 +107,43 @@ func newREST(resource schema.GroupResource, listKind schema.GroupVersionKind, st
 // Implement CategoriesProvider
 var _ rest.CategoriesProvider = &REST{}
 
+// List returns a list of items matching labels and field according to the store's PredicateFunc.
+func (e *REST) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	l, err := e.Store.List(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+
+	// Shallow copy ObjectMeta in returned list for each item. Native types have `Items []Item` fields and therefore
```

Comment:
does this also happen for objects returned in watch events? what about objects returned from Get?

Reply:
This looks suspicious as well: https://github.com/kubernetes/kubernetes/blob/70e45eccf27726f0e63dd1024924ccc7e2cd35a0/staging/src/k8s.io/apiserver/pkg/storage/cacher.go#L393

Reply:
Verified GET as well. As soon as you pass a ResourceVersion in the GetOptions, the cache is used (same for LIST) and the concurrent write happens.